connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala

```diff
@@ -242,17 +242,15 @@ class RemoteStreamingQuery(
   }
 
   override def exception: Option[StreamingQueryException] = {
-    val exception = executeQueryCmd(_.setException(true)).getException
-    if (exception.hasExceptionMessage) {
-      Some(
-        new StreamingQueryException(
-          // message maps to the return value of original StreamingQueryException's toString method
-          message = exception.getExceptionMessage,
-          errorClass = exception.getErrorClass,
-          stackTrace = exception.getStackTrace))
-    } else {
-      None
+    try {
+      // When setException is false, the server throws a StreamingQueryException
+      // to the client.
+      executeQueryCmd(_.setException(false))
+    } catch {
+      case e: StreamingQueryException => return Some(e)
```

Comment:
Thanks for the PR!
May I know if there are reasons why we want to do this throw way rather than have the client reconstruct the error? 

I remember letting the server throw will limit the output size of the exception to 2048 chars, which normally would be too small for the streaming query exceptions. I was thinking this method could mitigate this situation

Reply:
With https://github.com/apache/spark/pull/42377, error translation (via throw) not only reconstructs StreamingQueryException but also its cause exceptions with full message (via a additional RPC) so it is better